sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -559,6 +559,25 @@ case class DataSource(
       }
       globPath
     }.toSeq
+
+    if (checkFilesExist) {
+      val (filtered, filteredOut) = allGlobPath.partition { path =>
+        !InMemoryFileIndex.shouldFilterOut(path.getName)
+      }
+      if (filteredOut.nonEmpty) {
+        if (filtered.isEmpty) {
+          throw new AnalysisException(
+            "All path were ignored. The following path were ignored:\n" +
+              s"${filteredOut.mkString("\n  ")}")
+        } else {
+          logDebug(
+            "The following path were ignored:\n" +
```

Comment:
Nit: for performance, make this one interpolated string. If the line is too long make the variable `filteredOut` something shorter like `out`

Reply:
I fixed it.